File: Easy/3024. Type of Triangle/JavaScript/main.js
/**
 * @param {number[]} nums
 * @return {string}
 */
var triangleType = function (nums) {
  const [a, b, c] = nums;

  // Step 1: Check if it forms a valid triangle
  if (a + b <= c || a + c <= b || b + c <= a) {
    return "none";
  }

  // Step 2: Determine type
  if (a === b && b === c) {
    return "equilateral";
  } else if (a === b || a === c || b === c) {
    return "isosceles";
  } else {
    return "scalene";
  }
};

File: Easy/3024. Type of Triangle/Rust/main.rs
impl Solution {
    pub fn triangle_type(nums: Vec<i32>) -> String {
        let (a, b, c) = (nums[0], nums[1], nums[2]);

        // Step 1: Check if the three sides form a valid triangle
        if a + b <= c || a + c <= b || b + c <= a {
            return "none".to_string();
        }

        // Step 2: Determine the type of triangle
        if a == b && b == c {
            "equilateral".to_string()
        } else if a == b || a == c || b == c {
            "isosceles".to_string()
        } else {
            "scalene".to_string()
        }
    }
}
